src/RulesTest/Library/ControlShouldNotSupportTablePattern.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Axe.Windows.Core.Bases;
+using Axe.Windows.Core.Types;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Linq;
+using EvaluationCode = Axe.Windows.Rules.EvaluationCode;
+
+namespace Axe.Windows.RulesTest.Library
+{
+    [TestClass]
+    public class ControlShouldNotSupportTablePattern
+    {
+        private Axe.Windows.Rules.IRule Rule = new Axe.Windows.Rules.Library.ControlShouldNotSupportTablePattern();
+
+        [TestMethod]
+        public void NoTablePattern_Pass()
+        {
+            var e = new MockA11yElement();
+
+            Assert.AreEqual(EvaluationCode.Pass, Rule.Evaluate(e));
+        }
+
+        [TestMethod]
+        public void HasTablePattern_Error()
+        {
+            var e = new MockA11yElement();
+            e.Patterns.Add(new A11yPattern(e, PatternType.UIA_TablePatternId));
+
+            Assert.AreEqual(EvaluationCode.Error, Rule.Evaluate(e));
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ArgumentNull_Exception()
+        {
+            Rule.Evaluate(null);
```

Comment:
I've recently started to use a new format for expected exceptions, patterned after what is used in the TypeScript library. Certainly not required, but just for consideration. Remove the ExpectedException attribute from the test, then declare the expectation inside the function:

Assert.ThrowsException<ArgumentNullException)(() => Rule.Evaluate(null));

I personally prefer having the declaration of the expectation on the same line as the call that triggers the Exception

Reply:
I like it better too. I don't think I'll change it now though. Perhaps in the future as we alter tests we can change to that format.